compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -1340,16 +1340,32 @@ class OMR_EXTENSIBLE CodeGenerator
 
    TR::RealRegister **_unlatchedRegisterList; // dynamically allocated
 
-   bool alwaysUseTrampolines() { return _enabledFlags.testAny(AlwaysUseTrampolines); }
-   void setAlwaysUseTrampolines() {_enabledFlags.set(AlwaysUseTrampolines);}
-
    bool shouldBuildStructure() { return _enabledFlags.testAny(ShouldBuildStructure); }
    void setShouldBuildStructure() {_enabledFlags.set(ShouldBuildStructure);}
 
 
    bool enableRefinedAliasSets();
    void setEnableRefinedAliasSets() {_enabledFlags.set(EnableRefinedAliasSets);}
 
+   /**
+    * @brief Answers whether a trampoline is required for a direct call to reach a
+    *        target address from some source address.  This function should be
+    *        overridden by an architecture-specific implementation.
+    *
+    * @param[in] targetAddress : the absolute address of the call target
+    * @param[in] sourceAddress : the absolute address from which the call displacement
+    *              range is measured.  The meaning of this address is architecture
+                   specific.
+    *
+    * @return : true.  The default value is true because it should only result in
+    *              correct (but possibly suboptimal) behaviour.
+    */
+   bool directCallRequiresTrampoline(intptrj_t targetAddress, intptrj_t sourceAddress)
+      {
+      TR_ASSERT(0, "An architecture specialization of this function must be provided.");
```

Comment:
Sounds like a fatal assert. Can we make it as such?

Reply:
Yes, I'm ok with that suggestion.  However, the idea behind returning `true` was that it should force the user of this API to do the right thing because the target is not apparently reachable (i.e., its a fail safe).  A fatal assert, however, would make it more clear to an implementation that it should be overriding this function.